connector/connector.go

```diff
@@ -453,3 +453,16 @@ func NewFactory(cfgType component.Type, createDefaultConfig component.CreateDefa
 	}
 	return f
 }
+
+// MakeFactoryMap takes a list of connector factories and returns a map with factory type as keys.
+// It returns a non-nil error when there are factories with duplicate type.
+func MakeFactoryMap(factories ...Factory) (map[component.Type]Factory, error) {
```

Comment:
Not sure if other `MakeFactoryMap` functions do so but is it worth checking if the factory is `nil`?

Reply:
Since it's a variadic function, the input can be looped over safely, even if empty. Whether it should ever be empty is another question but it seems reasonable to let the caller handle that.